### PR TITLE
Fix $push definition in ArraySpec

### PR DESCRIPTION
**Pull request by dogan-acuris**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export type Spec<T, C extends CustomCommands<object> = never> =
   | (C extends CustomCommands<infer O> ? O : never);
 
 type ArraySpec<T, C extends CustomCommands<object>> =
-  | { $push: T }
+  | { $push: Array[T] }
```
Comment (kolodny) [inline index.d.ts:52]: Looks like you need to use `T[]` instead

File: index.d.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ export type Spec<T, C extends CustomCommands<object> = never> =
...
   | { $unshift: T }
```
Comment (kolodny) [inline index.d.ts:53]: Does this change need to be here too?

**Review by kolodny — COMMENTED** [review]
Looks good overall, is it possible to make a test for this. No big deal if it's not doable.

File: index.d.ts (hunk 1 above)
Comment (dogan-acuris) [reply index.d.ts:52]: ah yes sorry!

Comment (dogan-acuris) [reply index.d.ts:52]: Should I send another pull request with `T[]` ?

Comment (kolodny) [reply index.d.ts:52]: You can just make another commit with the change and push it to your `patch-1` branch it should be reflected in this PR

File: index.d.ts (hunk 2 above)
Comment (seansfkelley) [reply index.d.ts:53]: Yes.

File: index.d.ts (hunk 1 above)
Comment (seansfkelley) [reply index.d.ts:52]: My mistake, I somehow missed that push takes an array, not an item. This should indeed be `T[]`.

Please make sure the tests are updated to reflect this change!

**Comment by seansfkelley** [comment]
@kolodny #113 has the full set of required changes if you want to close this PR out instead.

**Comment by kolodny** [comment]
closed via #113 Thanks for the PR

**Pull request closed without merging**